pinecone/core/grpc/index_grpc.py

```diff
@@ -81,9 +82,38 @@ def __init__(self, index_name: str, channel=None, grpc_config: GRPCClientConfig
             "client-version": CLIENT_VERSION
         }
         self._endpoint_override = _endpoint_override
+
+        self.method_config = json.dumps(
+            {
+                "methodConfig": [
+                    {
+                        "name": [{"service": "VectorService.Upsert"}],
```

Comment:
Nice catches in this PR :)

Also nice that this gives us the opportunity to differentiate between reads vs writes in our retry strategy; maybe we should consider reducing the read default retries to a max of 3 attempts since they'll generally be more latency sensitive.

But, we should still allow users to override these values like they can today by passing a grpc_config with retry_config set to the GRPCIndex constructor.

Reply:
@benjaminran good idea. We'll add that to the task list of Client 3.0.

Reply:
Will we release this code before client 3.0? We shouldn't break backwards compatibility in the meantime

Reply:
Hmmm... I guess you're right.  
I'll add a way to either tweak the params or pass your own `method_config`